### PR TITLE
[EasyStandard] Add ecs sniff for test method names validation

### PR DESCRIPTION
**Pull request by BOB41K1987**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

File: packages/EasyStandard/src/Sniffs/Methods/TestMethodNameSniff.php [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use SlevomatCodingStandard\Helpers\NamespaceHelper;
+
+final class TestMethodNameSniff implements Sniff
+{
+    /**
+     * @var mixed[]
+     */
+    public $permitted = [
```
Comment (roman-eonx) [inline packages/EasyStandard/src/Sniffs/Methods/TestMethodNameSniff.php:16]: Let's rename to `allowed`.

File: packages/EasyStandard/tests/Sniffs/Methods/Fixtures/Correct/MethodNameConformWithPermittedPatterns.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+declare(strict_types=1);
+
+namespace EonX\EasyStandard\Tests\Sniffs\Methods\Fixtures\Correct;
+
+final class MethodNameConformWithPermittedPatterns
```
Comment (roman-eonx) [inline packages/EasyStandard/tests/Sniffs/Methods/Fixtures/Correct/MethodNameConformWithPermittedPatterns.php:6]: `Conforms`.

File: packages/EasyStandard/tests/Sniffs/Methods/Fixtures/Correct/AnotherNamespace/NamespaceDoesNotHaveForbiddenPatterns.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?php
+declare(strict_types=1);
```
Comment (roman-eonx) [inline packages/EasyStandard/tests/Sniffs/Methods/Fixtures/Correct/AnotherNamespace/NamespaceDoesNotHaveForbiddenPatterns.php:2]: An empty line is missing. Please check the ECS Github action.

File: packages/EasyStandard/tests/Sniffs/Methods/Fixtures/Correct/MethodNameConformWithPermittedPatterns.php (hunk 2 above)
Comment (BOB41K1987) [reply packages/EasyStandard/tests/Sniffs/Methods/Fixtures/Correct/MethodNameConformWithPermittedPatterns.php]: Done.

File: packages/EasyStandard/src/Sniffs/Methods/TestMethodNameSniff.php (hunk 1 above)
Comment (BOB41K1987) [reply packages/EasyStandard/src/Sniffs/Methods/TestMethodNameSniff.php]: Done.

**Review by roman-eonx — APPROVED** [review]

**Review by natepage — APPROVED** [review]

**Pull request merged**